DSESwift/SensorDataProxy.swift

```diff
@@ -0,0 +1,388 @@
+    * @param endTime        End time of the data series to be deleted.
+    *                       When endTime is null, all data from startTime till now will be removed.
+    *                       When both startTime and endTime are null, all data will be removed.
+    */
+    func deleteSensorData(sourceName sourceName: String, sensorName: String, startDate: NSDate? = nil, endDate: NSDate? = nil) throws {
+        if (!isSessionIdSet()){ throw ProxyError.SessionIdNotSet }
+        if (!isStartDateEarlierThanEndDate(startDate, endDate)){ throw ProxyError.InvalidQuery }
+        
+        var params = Dictionary<String, AnyObject>()
+        if startDate != nil { params["start_time"] = JSONUtils.stringify(startDate!.timeIntervalSince1970)}
+        if endDate != nil { params["end_time"] = JSONUtils.stringify(endDate!.timeIntervalSince1970)}
+        let result = Just.delete(getSensorDataUrl(sourceName, sensorName), params: params, headers: getHeaders())
+        if let error = checkStatusCode(result.statusCode!, successfulCode: 201){
+            throw error
+        }
+    }
+    
+ 
+    // MARK: Helper functions
+    
+    /**
+    * Helper function to create a JSONObject with the following structure:
+    *
+    *     {
+    *       source_name: string,
+    *       sensor_name, string,
+    *       data: [
+    *         {date: number, value: JSON},
+    *         ...
+    *       ]
+    *     }
+    *
+    * This helper function can be used to prepare the data for putSensorData.
+    * @param sourceName     The source name, for example "sense-ios",
+    *                       "sense-android", "fitbit", ...
+    * @param sensorName     The sensor name, for example "accelerometer"
+    * @param data           Array with data points, structured as `[{date: long, value: JSON}, ...]`
+    * @param meta           Optional field to store meta information. Can be left null
+    */
+    static func createSensorDataObject (sourceName sourceName: String, sensorName: String, data: Array<AnyObject>, meta: Dictionary<String, AnyObject>? = nil) -> Dictionary<String, AnyObject> {
+        var sensorData = Dictionary<String, AnyObject>()
+        sensorData["source_name"] = sourceName;
+        sensorData["sensor_name"] = sensorName;
+        if (meta != nil) {
+            sensorData["meta"] = meta!
+        }
+        sensorData["data"] = data;
+        
+        return sensorData;
+    }
+    
+    /**
+    * Returns NSData containing JSONArray. It perform JSONSerizlization on the give Array.
+    * Don't forget to validate the Array by calling NSJSONSerialization.isValidJSONObject before using this method.
+    */
+    private func serializeArrayToJSONArray(sensorsData: Array<AnyObject>) -> NSData?{
+        var body: NSData?
+        do{
+            body = try NSJSONSerialization.dataWithJSONObject(sensorsData, options: NSJSONWritingOptions(rawValue: 0))
+        }catch{
+            debugPrint("Wrong format on JsonSerialization process.")
```

Comment:
@kanekin won't this mean the function will still continue and return body when there is an error?